lib/Controller/SettingsController.php

```diff
@@ -47,20 +50,64 @@ class SettingsController extends Controller {
 	private $providerService;
 	/** @var ICrypto */
 	private $crypto;
+	/** @var IClientService */
+	private $clientService;
+	/** @var ILogger */
+	private $logger;
 
 	public function __construct(
 		IRequest $request,
 		ProviderMapper $providerMapper,
 		ID4MeService $id4meService,
 		ProviderService $providerService,
-		ICrypto $crypto
-		) {
+		ICrypto $crypto,
+		IClientService $clientService,
+		LoggerInterface $logger
+	) {
 		parent::__construct(Application::APP_ID, $request);
 
 		$this->providerMapper = $providerMapper;
 		$this->id4meService = $id4meService;
 		$this->providerService = $providerService;
 		$this->crypto = $crypto;
+		$this->clientService = $clientService;
+		$this->logger = $logger;
+	}
+
+	public function isDiscoveryEndpointValid($url) {
+		$result = [
+			'isReachable' => false,
+			'missingFields' => [],
+		];
+
+		try {
+			$client = $this->clientService->newClient();
+			$response = $client->get($url);
+			$httpCode = $response->getStatusCode();
+			$body = $response->getBody();
+
+			// Check if the request was successful
+			if ($httpCode == 200 && !empty($body)) {
+				$result['isReachable'] = true;
+				$data = json_decode($body, true);
+
+				// Check for required fields as defined in: https://openid.net/specs/openid-connect-discovery-1_0.html
+				$requiredFields = [
+					'issuer', 'authorization_endpoint', 'token_endpoint', 'jwks_uri',
+					'response_types_supported', 'subject_types_supported', 'id_token_signing_alg_values_supported',
+				];
+				
+				foreach ($requiredFields as $field) {
+					if (!isset($data[$field])) {
+						$result['missingFields'][] = $field;
+					}
+				}
+			}
+		} catch (Exception $e) {
+			$this->logger->error('Error - discovery endpoint validation: ' . $e);
```

Comment:
```suggestion
			$this->logger->error('Discovery endpoint validation error', ['exception' => $e]);
```
So we get more info on the exception.